### PR TITLE
util conversion: wave2rgb refactored to become more readable and redu…

### PR DESCRIPTION
**Pull request by tepals**
…ce the number of lines.

### CODE REVIEW

File: src/odemis/util/conversion.py [hunk 1]
```diff
@@ -31,49 +31,36 @@
...
-        r = 1
-        g = -(w - 645) / (645 - 580)
-        b = 0
-    elif 645 <= w <= 780:
-        r = 1
-        g = 0
-        b = 0
+    w *= 1e9
+
+    x = lambda w, l, h: round((w - h) / abs((h - l)) * 255)  # maps w between 0 and +/- 1
```
Comment (pieleric) [inline src/odemis/util/conversion.py:49]: Could you give this function a better name? You don't have to limit yourself to one character ;-)
Also you could add some explanation on what it returns. It seems sometimes it returns negative values.
Moreover, l and h sounds like meaning low/high. But when low < high, it returns negative values... which is odd.

**Review by pieleric — CHANGES_REQUESTED** [review]

File: src/odemis/util/conversion.py [hunk 2]
```diff
@@ -31,49 +31,36 @@
...
-        b = 0
-    elif 580 <= w < 645:
-        r = 1
-        g = -(w - 645) / (645 - 580)
-        b = 0
-    elif 645 <= w <= 780:
-        r = 1
-        g = 0
-        b = 0
+    w *= 1e9
```
Comment (Kleijwegt) [inline src/odemis/util/conversion.py:47]: Could you add a comment that we go from wavelength in meters to wavelength in nm?

File: src/odemis/util/conversion.py [hunk 3]
```diff
@@ -31,49 +31,36 @@
 import re
 import yaml
 
-# Inspired by code from:
-# http://codingmess.blogspot.nl/2009/05/conversion-of-wavelength-in-nanometers.html
-# based on:
-# http://www.physics.sfasu.edu/astro/colour/spectra.html
-def wave2rgb(wavelength):
+
+def wave2rgb(w):
```
Comment (Kleijwegt) [inline src/odemis/util/conversion.py:35]: I think we should also update the name of this function to wavelenght2rgb

**Review by Kleijwegt — CHANGES_REQUESTED** [review]

**Comment by tepals** [comment]
I did not mean to add the formatting changes to conversion.py. but since there are not too many I'll leave them (if you guys don't mind).

File: src/odemis/util/conversion.py [hunk 4]
```diff
@@ -31,49 +31,38 @@
...
-    return (3-tupe int in 0..255): RGB value
+    Convert a wavelength into a (r, g, b) value
+    wavelength (0 < float): wavelength in m
+    return (3-tuple int in 0..255): RGB value
+
+    Notes:
+    Inspired by code from:
+    http://codingmess.blogspot.nl/2009/05/conversion-of-wavelength-in-nanometers.html
+    based on:
+    http://www.physics.sfasu.edu/astro/color/spectra.html
```
Comment (Kleijwegt) [inline src/odemis/util/conversion.py:45]: if this link doesn't work maybe we should hunt down a working one or write a note that it doesn't work.

**Review by Kleijwegt — APPROVED** [review]

File: src/odemis/util/conversion.py (hunk 4 above)
Comment (tepals) [reply src/odemis/util/conversion.py:45]: I already did, and this is the working one

**Review by pieleric — APPROVED** [review]

**Pull request merged**